content/dev/plugins/help.md

```diff
@@ -0,0 +1,111 @@
+
+```
+<?php
+class zcObserverMyPluginHelp extends base {
+  function __construct() {
+    $this->attach($this, array('NOTIFIER_PLUGIN_HELP_PAGE_URL_LOOKUP'));
+  }
+
+  function update(&$class, $eventID, $page, &$help_page) {
+     if ($eventID == 'NOTIFIER_PLUGIN_HELP_PAGE_URL_LOOKUP') {
+        if ($page == FILENAME_MYPLUGIN_NAME) {
+           $help_page = "help/myplugin.html"; 
+        }
+     }
+  }
+}
+```
+
+Then create a help file in `YOURADMIN/help/myplugin.html` (or `YOURADMIN/help/myplugin.php` if you need to execute code) to store your help text.
+
+If you are not running Zen Cart 1.5.7 or higher, you will need to pull in [this PR](https://github.com/zencart/zencart/commit/bc195baf258c11b73f29de41020e1c0505e4d462) to your cart to get admin observers to autoload. 
+
+## Help for Modules (Shipping, Payment or Order Totals) 
+
+TBD - waiting on PR 4246 to be merged. 
+
+## Examples of PHP Help Files 
+
+Quite often plain text help (i.e. a `.html` file) is perfectly adequate, but if you need to make reference to the values of settings, pull in files, or do other dynamic operations, you'll want to use a `.php` help file.
+
+### Setup 
+To have access to database settings, functions, and everything else, you need to pull in the configure file.  This is a two step process since you're not at the top level of either admin or the storefront.  First, `chdir` to the admin (or catalog) folder, then pull it in.
+
+```
+<?php 
+chdir("..");
+require 'includes/application_top.php';
+?>
+```
+
+You still have to account for being in a subfolder when building links - see examples below. 
+
+### Displaying and Linking to Config values in Core menus 
+
+Suppose you created a database entry with configuration key SPECIAL_FEE in the Admin > Configuration > My Store section.  Here's how you could access it and reference it: 
+
+```
+<?php
+echo 'Current Special Fee is $' . SPECIAL_FEE . '.<br><br>';
+echo 'To change this value, go to <a href="../configuration.php?gID=1" target="_blank">Admin > Configuration > My Store</a> <br>';
+?>
+```
+
+### Displaying and Linking to Config values in Plugin menus 
+
+If you don't know what the group id is, you have to query it. 
+Rather than querying the group title (a text string that could change), query the configuration group of a specific key in that group. 
+
+```
+<?php 
+$group = $db->Execute("SELECT * FROM " . TABLE_CONFIGURATION . " WHERE configuration_key = 'PO_SEND_PACKING_LISTS'");
+$gID = $group->fields['configuration_group_id'];
+echo 'Settings for this plugin are configured in <a href="'  . HTTPS_CATALOG_SERVER . DIR_WS_ADMIN . '../configuration.php?gID=' . $gID . '" target="_blank">Admin > Configuration > Dropship Purchase Orders</a> <br>';
+?>
+```
+
+### Pulling in a define page 
+
+Using a custom built [define page](/user/template/define_pages/) to store content means the storeowner can update the help independently without needing a developer.
```

Comment:
Do you want me to rewrite this with EZPages?

Reply:
Well, ez-pages isn't intended for internal help/documentation.

While I dislike encouraging Define Pages because of all the complexity it requires for catalog-side use and because it's entirely wrongly named, and other reasons, it could indeed "work" in this situation. 

But storeowners will potentially experience confusion about which "define pages" are actually storefront-related vs only admin-related-and-only-for-help-and-documentation because they're worlds apart in terms of their intended purpose.

Reply:
I'd like to offer something for storeowners to use for making notes and recording procedures.  What do you suggest?

Reply:
> I'd like to offer something for storeowners to use for making notes and recording procedures. What do you suggest?

- The Admin Notes mod does that, both generally, and per-customer. It's a good mid-step to a CRM. Zen Cart is not intended to be a CRM.

- A shared Google Drive folder (or whatever cloud provider you prefer) is a great way to let staff collaborate on Procedures, Instructions, Notes, and takes care of all access-control as well.

Reply:
Fixed on next checkin.